go-controller/pkg/util/ovn.go

```diff
@@ -67,3 +69,9 @@ func GetDatapathUUID(datapathName string) (string, error) {
 	}
 	return datapath, nil
 }
+
+func PlatformTypeIsEgressIPCloudProvider() bool {
```

Comment:
so it looks like you use this function to determine if the new controller should be enabled, but this is seems incorrect to me. The platform type has nothing to do with if you want to enable this controller. That should be a separate config flag.

Reply:
> The platform type has nothing to do with if you want to enable this controller. 

It's completely tied to it. On public clouds egress IP will never work without that controller.

Reply:
sure i agree its tied to it, but the platform you are using shouldn't decide if egress ip with this controller is used. However, I just realized we have config.OVNKubernetesFeature.EnableEgressIP to enable egressIP. So I guess user configuring that and the platform is enough to say ok the intention is to enable this behavior.

Would it be possible for someone to use egress IP without the controller? For example, what if someone reserved an IP on a cloud and assigned it to a node manually, and then labeled just that node as egress IP with that IP. That should work right? Obviously, no HA there.

Reply:
> Would it be possible for someone to use egress IP without the controller? 

Yes, someone could do what you say, but that would involve _a lot_ of work. Given the normal operations of egress IP (specifically that they can move whenever the egress IP controller detects a node as "unavailable"), doing that manually without this controller would make that persons life very very miserable. 

> For example, what if someone reserved an IP on a cloud and assigned it to a node manually, and then labeled just that node as egress IP with that IP

Yes, theoretically that would work.